src/routes/accounts/counterfactual-safes/counterfactual-safes.controller.spec.ts

```diff
@@ -584,4 +584,337 @@ describe('CounterfactualSafesController', () => {
+      expect(accountsRepository.getAccount).not.toHaveBeenCalled();
+      expect(
+        counterfactualSafesDataSource.deleteCounterfactualSafesForAccount,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('returns 403 is token it not yet valid', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const chain = chainBuilder().build();
+      const authPayloadDto = authPayloadDtoBuilder()
+        .with('chain_id', chain.chainId)
+        .with('signer_address', address)
+        .build();
+      const accessToken = jwtService.sign({
+        ...authPayloadDto,
+        nbf: faker.date.future(),
+      });
+
+      await request(app.getHttpServer())
+        .delete(`/v1/accounts/${address}/counterfactual-safes`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(403);
+
+      expect(() => jwtService.verify(accessToken)).toThrow('jwt not active');
+      expect(accountsRepository.getAccount).not.toHaveBeenCalled();
+      expect(
+        counterfactualSafesDataSource.deleteCounterfactualSafesForAccount,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('returns 403 if token has expired', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const chain = chainBuilder().build();
+      const authPayloadDto = authPayloadDtoBuilder()
+        .with('chain_id', chain.chainId)
+        .with('signer_address', address)
+        .build();
+      const accessToken = jwtService.sign({
+        ...authPayloadDto,
+        exp: faker.date.past(),
+      });
+
+      expect(() => jwtService.verify(accessToken)).toThrow('jwt expired');
+      await request(app.getHttpServer())
+        .delete(`/v1/accounts/${address}/counterfactual-safes`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(403);
+
+      expect(accountsRepository.getAccount).not.toHaveBeenCalled();
+      expect(
+        counterfactualSafesDataSource.deleteCounterfactualSafesForAccount,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('returns 403 if signer_address is not a valid Ethereum address', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const chain = chainBuilder().build();
+      const authPayloadDto = authPayloadDtoBuilder()
+        .with('chain_id', chain.chainId)
+        .with('signer_address', faker.string.sample() as `0x${string}`)
+        .build();
+      const accessToken = jwtService.sign(authPayloadDto);
+
+      await request(app.getHttpServer())
+        .delete(`/v1/accounts/${address}/counterfactual-safes`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .expect(403);
+
+      expect(accountsRepository.getAccount).not.toHaveBeenCalled();
+      expect(
+        counterfactualSafesDataSource.deleteCounterfactualSafesForAccount,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('returns 403 if chain_id is not a valid chain ID', async () => {
```

Comment:
What do you think about adding another test case that ensure the token is valid if signed on a different chain?

Reply:
I think that is not applicable in this case as the route URL does not contain a Chain ID, right?

Reply:
If "signed in" on another chain, we still want to return counterfactual setups on others. I'd say it's more of a "preventative" test to ensure failure if this is altered in the future. What do you think?

Reply:
I was giving this another thought and cannot find how to implement the desired test. I mean, `chainId` is only used to compose the `authPayload`:
```
      const chain = chainBuilder().build();
      const authPayloadDto = authPayloadDtoBuilder()
        .with('chain_id', chain.chainId)
        .with('signer_address', address)
        .build();
```

And the CF Safes datasource deletion is mocked:
```
counterfactualSafesDataSource.deleteCounterfactualSafesForAccount.mockResolvedValue();
```

So there's no link between the `AuthPayload.chainId` and the `chainId` of the Counterfactual Safes being deleted. If this is altered in the future (i.e.: a correspondence between both `chainId` properties is required), we would need to add a second argument to the datasource function. I've added an assertion on the datasource call in [3d81c8c](https://github.com/safe-global/safe-client-gateway/pull/1795/commits/3d81c8cfa6b5291f0ced0689adb9419856839261)

Please let me know if you think there's another way to test it! Thanks!